mitosheet/mitosheet/step_performers/ai_transformation.py

```diff
@@ -0,0 +1,81 @@
+
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+
+import pandas as pd
+from time import perf_counter
+from typing import Any, Dict, List, Optional, Set, Tuple
+from mitosheet.code_chunks.code_chunk import CodeChunk
+from mitosheet.code_chunks.ai_transformation_code_chunk import AITransformationCodeChunk
+
+from mitosheet.state import State
+from mitosheet.step_performers.step_performer import StepPerformer
+from mitosheet.step_performers.utils import get_param
+from mitosheet.ai.recon import exec_and_get_new_state_and_last_line_expression_value
+
+class AITransformationStepPerformer(StepPerformer):
+    """
+    Allows you to execute an arbitrary chunk of code, generated by an AI.
+    """
+
+    @classmethod
+    def step_version(cls) -> int:
+        return 1
+
+    @classmethod
+    def step_type(cls) -> str:
+        return 'ai_transformation'
+
+    @classmethod
+    def execute(cls, prev_state: State, params: Dict[str, Any]) -> Tuple[State, Optional[Dict[str, Any]]]:
+        user_input: str = get_param(params, 'user_input')
+        prompt_version: str = get_param(params, 'prompt_version')
+        prompt: str = get_param(params, 'prompt')
+        completion: str = get_param(params, 'completion')
+        edited_completion: str = get_param(params, 'edited_completion')
+
+        pandas_start_time = perf_counter()
+        post_state, last_line_value = exec_and_get_new_state_and_last_line_expression_value(prev_state, edited_completion)
+        pandas_processing_time = perf_counter() - pandas_start_time
+
+        # If the last line value is a primitive, we return it as a result for the frontend
+        result_last_line_value = None
+        if isinstance(last_line_value, str) or isinstance(last_line_value, bool) or isinstance(last_line_value, int) or isinstance(last_line_value, float):
```

Comment:
it could be other things than these, like a list of names, for example.

Reply:
yeah. The most common is probably going to be a series. I'm down to support these + a series for the start, and log if it's something else.